pkg/beacon/relay/node.go

```diff
@@ -157,6 +158,61 @@ func (n *Node) ForwardSignatureShares(groupPublicKeyBytes []byte) {
 	n.netProvider.BroadcastChannelForwarderFor(name)
 }
 
+// ResumeSigningIfEligible enables a client to rejoin the ongoing signing process
+// after it was crashed or restarted and if it belongs to the signing group.
+func (n *Node) ResumeSigningIfEligible(
+	relayChain relayChain.Interface,
+	signing chain.Signing,
+) {
+	isEntryInProgress, err := relayChain.IsEntryInProgress()
+	if err != nil {
+		logger.Errorf(
+			"failed checking if an entry is in progress: [%v]",
+			err,
+		)
+		return
+	}
+
+	if isEntryInProgress {
+		previousEntry, err := relayChain.CurrentRequestPreviousEntry()
+		if err != nil {
+			logger.Errorf(
+				"failed getting previous entry on current request: [%v]",
+				err,
+			)
+			return
+		}
+		entryStartBlock, err := relayChain.CurrentEntryStartBlock()
+		if err != nil {
+			logger.Errorf(
+				"failed getting a current relay entry start block: [%v]",
+				err,
+			)
+			return
+		}
+		groupPublicKey, err := relayChain.GroupPublicKey()
+		if err != nil {
+			logger.Errorf(
+				"failed getting a group public key: [%v]",
+				err,
+			)
+			return
+		}
+
+		logger.Infof(
+			"client is atempting to rejoin the current signing group [0x%x]",
```

Comment:
I'd rather say `the current signing process` because the client is in the signing group, it just wasn't active.

Reply:
> because the client is in the signing group, it just wasn't active.

I don't think we know this at this point yet. Looks like we validate if it belongs to the currently signing group down the stream [here](https://github.com/keep-network/keep-core/blob/master/pkg/beacon/relay/relay.go#L108)

But I can say `process`, kinda gives the same idea on what is happening.